src/test/compile-fail/issue-68523.rs

```diff
@@ -0,0 +1,6 @@
+// compile-flags: --edition 2018
+// error-pattern: `main` function is not allowed to be `async`
+
+async fn main() -> Result<i32, ()> {
+    Ok(1)
+}
```

Comment:
Could you move the tests to `src\test\ui\async-await` and use `//~ ERROR ...` annotations instead of `// error-pattern: ...`?

Reply:
Sure sounds good